src/map/battlefield.cpp

```diff
@@ -235,33 +235,25 @@ uint8 CBattlefield::getPlayerMainLevel()
 }
 
 void CBattlefield::capPlayerToBCNM()
-{ //adjust player's level to the appropriate cap and remove buffs
+{
+    //adjust player's level to the appropriate cap and remove buffs
     if (m_PlayerList.size() == 0)
     {
         ShowWarning("battlefield:getPlayerMainLevel - No players in battlefield!\n");
         return;
     }
-    uint8 cap = getLevelCap();
+
+    uint8 cap = std::clamp<uint8>(getLevelCap(), 0, 99);
```

Comment:
cap should be between 1 and 99 instead of 0 and 99?

Reply:
no because a zero cap is valid.

99 cap gets used to strip buff 0 gets used for uncapped that doesn't strip buffs